CORE-6067: Setting up network scripts (#2177)
corda/corda-runtime-os: tools/plugins/mgm/README.md

      endpointProtocol: 10
    - name: "C=GB, L=London, O=Member2"
```

# Setup Kubernetes networks

This is a sub-command under the `mgm` plugin for setting up Kubernetes networks. 

Running `setupNetwork` with the name of the networks to create. For example:
```shell
./corda-cli.sh mgm setupNetwork demo-network-one demo-network-two
```
By default, it will use the latest released tag. Change it using the `--baseImage` option. It will delete any
existing network with that name. Use the `--help` to view all the other options.

# Onboard a member to an existing network
This is a sub-command under the `mgm` plugin for on-boarding a member (MGM or standard member) into a running network.

To run the network either use the `setupNetwork` command or run a combine worker locally ([see here](../../../applications/workers/release/combined-worker/README.md)).

## Onboard an MGM member to an existing network
This is a sub-command under the `onboard` sub-command to on board a new MGM member (and create a new group).

To onboard on a Kubernetes network use the network name as parameter. By default, it will try to on board on a combined worker.

> Use the `--save-group-policy-as` to indicate where to save the MGM group policy file (that can be used to create CPIs - [see here](../package/README.md))

Reviewer: Is there a default place for this and can we provide an example command?

An example of on-boarding an MGM can be:
```shell
./corda-cli.sh mgm onboard mgm demo-network-one  --x500-name='O=MGM, L=London, C=GB'
```
Use the `--help` to view all the other options.

See [here](https://github.com/corda/corda-runtime-os/wiki/MGM-Onboarding) for details on how to do it manually.

## Onboard a standard member to an existing network
This is a sub-command under the `onboard` sub-command to on board a new member to an existing group.

To onboard on a Kubernetes network use the network name as parameter. By default, it will try to on board on a combined worker.
To decide which CPI to use, there are three options:
* If you know the CPI hash, you can use it with the `--cpi-hash` option
* If you have the CPI file (for example, from the [package command](../package/README.md)), you can use it with the `--cpi-file` option.

> * If you have a CPB and a group policy file (from the `onboard mgm` command), you can use the `--cpb-file` and `--group-policy-file` option. This will create an unsigned CPI and save it in your home directory.

Reviewer: Here as well would be good to see examples for everything.

      endpointProtocol: 10
    - name: "C=GB, L=London, O=Member2"
```

> # Setup Corda cluster in Kubernetes

Reviewer 2: It could be good to explicitly say that this is a dev tool only since it uses the dev CA and `admin`/`admin` log-in for RPC commands.

This is a sub-command under the `mgm` plugin for setting up Kubernetes networks. 

> Running `setupCluster` with the name of the networks to create. For example:

Reviewer 2: ```suggestion
Running `setupCluster` with the name of the clusters to create. For example:
```

```shell

> ./corda-cli.sh mgm setupCluster demo-network-one demo-network-two

Reviewer 2: I know these names could be anything but it'd be good to keep it consistent with the network -> cluster changes.
```suggestion
./corda-cli.sh mgm setupCluster demo-cluster-one demo-cluster-two
```

```
By default, it will use the latest released tag. Change it using the `--baseImage` option. It will delete any

> existing network with that name. Use the `--help` to view all the other options.

Reviewer 2: ```suggestion
existing cluster with that name. Use the `--help` to view all the other options.
```

> # Onboard a member to an existing network

Reviewer 2: ```suggestion
# Onboard a member to an existing corda cluster
```

> This is a sub-command under the `mgm` plugin for on-boarding a member (MGM or standard member) into a running network.

Reviewer 2: ```suggestion
This is a sub-command under the `mgm` plugin for on-boarding a member (MGM or standard member) into a running corda cluster.
```

> To run the network either use the `setupNetwork` command or run a combine worker locally ([see here](../../../applications/workers/release/combined-worker/README.md)).

Reviewer 2: ```suggestion
To run the network either use the `setupCluster` command or run a combine worker locally ([see here](../../../applications/workers/release/combined-worker/README.md)).
```

> ## Onboard an MGM member to an existing network

Reviewer 2: ```suggestion
## Onboard an MGM member to an existing corda cluster
```

> This is a sub-command under the `onboard` sub-command to on board a new MGM member (and create a new group).

Reviewer 2: ```suggestion
This is a sub-command under the `onboard` sub-command to onboard a new MGM member (and create a new group).
```

> To onboard on a Kubernetes network use the network name as parameter. By default, it will try to on board on a combined worker.

Reviewer 2: I could be wrong but I think the k8s terminology is cluster also instead of network.
```suggestion
To onboard on a Kubernetes cluster use the cluster name as parameter. By default, it will try to onboard on a combined worker.
```

By default, the command will save the group policy file into `~/.corda/gp/groupPolicy.json` (and will overwrite any existing group policy file there).
Use the `--save-group-policy-as` to indicate another location to save the MGM group policy file (that can be used to create CPIs - [see here](../package/README.md))

Examples of on-boarding an MGM can be:
```shell

> ./corda-cli.sh mgm onboard mgm demo-network-one  --x500-name='O=MGM, L=London, C=GB'

Reviewer 2: ```suggestion
./corda-cli.sh mgm onboard mgm demo-cluster-one --x500-name='O=MGM, L=London, C=GB'
```

./corda-cli.sh mgm onboard mgm --save-group-policy-as /tmp/groupPolicy.json
```
Use the `--help` to view all the other options and defaults.

See [here](https://github.com/corda/corda-runtime-os/wiki/MGM-Onboarding) for details on how to do it manually.

> ## Onboard a standard member to an existing network

Reviewer 2: ```suggestion
## Onboard a standard member to an existing cluster
```

> This is a sub-command under the `onboard` sub-command to on board a new member to an existing group.

Reviewer 2: ```suggestion
This is a sub-command under the `onboard` sub-command to onboard a new member to an existing group.
```

> To onboard on a Kubernetes network use the network name as parameter. By default, it will try to on board on a combined worker.

Reviewer 2: ```suggestion
To onboard on a Kubernetes cluster use the corda cluster name as parameter. By default, it will try to on board on a combined worker.
```

To decide which CPI to use, there are three options:
* If you know the CPI hash, you can use it with the `--cpi-hash` option
* If you have the CPI file (for example, from the [package command](../package/README.md)), you can use it with the `--cpi-file` option.
* If you have a CPB and a group policy file (from the `onboard mgm` command), you can use the `--cpb-file` and `--group-policy-file` option. This will create an unsigned CPI and save it in your home directory.

Few examples of on-boarding a member can be:
```shell

> ./corda-cli.sh mgm onboard member demo-network-two --x500-name='O=Alice, L=London, C=GB' --cpb-file ~/corda-runtime-os/testing/cpbs/chat/build/libs/*.cpb

Reviewer 2: ```suggestion
./corda-cli.sh mgm onboard member demo-cluster-two --x500-name='O=Alice, L=London, C=GB' --cpb-file ~/corda-runtime-os/testing/cpbs/chat/build/libs/*.cpb
```

> ./corda-cli.sh mgm onboard member demo-network-one --cpi-file /tmp/calculator.cpi

Reviewer 2: ```suggestion
./corda-cli.sh mgm onboard member demo-cluster-one --cpi-file /tmp/calculator.cpi
```
